net/FlatBuffers/Table.cs

```diff
@@ -83,15 +84,15 @@ protected int __vector(int offset)
         }
 
         // Initialize any Table-derived type to point to the union at the given offset.
-        protected TTable __union<TTable>(TTable t, int offset) where TTable : Table
+        public T __union<T>(int offset) where T : struct, Accessor
```

Comment:
This is more in line with what you'd expect. Traverse largely the same (as it re-uses most of the objects), Encode is improved and Decode is about the same (as you said, 'Decode' does nothing of note).